tests/components/peco/test_sensor.py

```diff
@@ -0,0 +1,256 @@
+        json={"data": {"interval_generation_data": "data/TEST"}},
+    )
+    aioclient_mock.get(
+        "https://kubra.io/data/TEST/public/reports/a36a6292-1c55-44de-a6a9-44fedf9482ee_report.json",
+        status=500,
+        json={"error": "Internal Server Error"},
+    )
+
+    config_entry = MockConfigEntry(domain=DOMAIN, data=COUNTY_ENTRY_DATA)
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert hass.data[DOMAIN]
+
+    assert "Error getting PECO outage counter data" in caplog.text
+
+
+async def test_bad_json(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, caplog
+):
+    """Test if it raises an error when there is bad JSON."""
+    aioclient_mock.get(
+        "https://kubra.io/stormcenter/api/v1/stormcenters/39e6d9f3-fdea-4539-848f-b8631945da6f/views/74de8a50-3f45-4f6a-9483-fd618bb9165d/currentState?preview=false",
+        json={"data": {}},
+    )
+
+    config_entry = MockConfigEntry(domain=DOMAIN, data=COUNTY_ENTRY_DATA)
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert hass.data[DOMAIN]
+
+    assert "ConfigEntryNotReady" in caplog.text
+
+
+async def test_total_http_error(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, caplog
+):
+    """Test if it raises an error when there is an HTTP error."""
+    aioclient_mock.get(
+        "https://kubra.io/stormcenter/api/v1/stormcenters/39e6d9f3-fdea-4539-848f-b8631945da6f/views/74de8a50-3f45-4f6a-9483-fd618bb9165d/currentState?preview=false",
+        json={"data": {"interval_generation_data": "data/TEST"}},
+    )
+    aioclient_mock.get(
+        "https://kubra.io/data/TEST/public/reports/a36a6292-1c55-44de-a6a9-44fedf9482ee_report.json",
+        status=500,
+        json={"error": "Internal Server Error"},
+    )
+
+    config_entry = MockConfigEntry(domain=DOMAIN, data=MOCK_ENTRY_DATA)
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert hass.data[DOMAIN]
+
+    assert "Error getting PECO outage counter data" in caplog.text
+
+
+async def test_total_bad_json(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, caplog
+):
+    """Test if it raises an error when there is bad JSON."""
+    aioclient_mock.get(
+        "https://kubra.io/stormcenter/api/v1/stormcenters/39e6d9f3-fdea-4539-848f-b8631945da6f/views/74de8a50-3f45-4f6a-9483-fd618bb9165d/currentState?preview=false",
+        json={"data": {}},
+    )
+
+    config_entry = MockConfigEntry(domain=DOMAIN, data=MOCK_ENTRY_DATA)
+    config_entry.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert hass.data[DOMAIN]
+
+    assert "ConfigEntryNotReady" in caplog.text
```

Comment:
This is not a correct check, this checks if the stack trace ended up in the logs. But in fact, we need to check the entry hasn't been set up. Please check the config entry state instead.